test/integration/IntegrationTest.cpp

```diff
@@ -65,23 +65,46 @@ IntegrationTest::IntegrationTest()
       _sslDtls(nullptr),
       _network(transport::createRtcePoll()),
       _pacer(10 * utils::Time::ms),
-      _instanceCounter(0)
+      _instanceCounter(0),
+      _numWorkerThreads(getNumWorkerThreads())
 {
-    for (size_t threadIndex = 0; threadIndex < std::thread::hardware_concurrency(); ++threadIndex)
-    {
-        _workerThreads.push_back(std::make_unique<jobmanager::WorkerThread>(*_jobManager));
-    }
 }
 
 void IntegrationTest::SetUp()
 {
+
+    if (__has_feature(address_sanitizer) || __has_feature(thread_sanitizer))
+    {
+        GTEST_SKIP();
+    }
+#if !ENABLE_LEGACY_API
```

Comment:
Is this something we do now only? I mean the tests contain both legacy and SMB API.

Reply:
This macros was present in all the tests, basically disabling tests if the build was not done with the _legaci API_ support. So this changes nothing. All tests would be run, but only on the build configs that:
a) does not use thread/address sanitizers
b) supports _legacy API_.

In future, when the _legacy API_ will be deprecated, we'll remove certain tests, refactor others and remove this check at all.